libweston/backend-rdp/rdprail.c

```diff
@@ -4164,68 +4539,68 @@ rdp_rail_notify_app_list(void *rdp_backend, struct weston_rdprail_app_list_data
 	rdp_debug(b, "    appProvider: %s\n", app_list_data->appProvider);
 
 	if (app_list_data->deleteAppId) {
-		RDPAPPLIST_DELETE_APPLIST_PDU deleteAppList = {};
-		assert(app_list_data->appProvider == NULL); // provider must be NULL.
-		deleteAppList.flags = RDPAPPLIST_FIELD_ID;
-		if (app_list_data->appId == NULL || // appId is required.
-			!utf8_string_to_rail_string(app_list_data->appId, &deleteAppList.appId))
+		RDPAPPLIST_DELETE_APPLIST_PDU delete_app_list = {};
+
+		assert(app_list_data->appProvider == NULL);
+		delete_app_list.flags = RDPAPPLIST_FIELD_ID;
+		if (app_list_data->appId == NULL ||
+		    !utf8_string_to_rail_string(app_list_data->appId, &delete_app_list.appId))
 			goto Exit_deletePath;
 
-		if (app_list_data->appGroup && // appGroup is optional.
-			utf8_string_to_rail_string(app_list_data->appGroup, &deleteAppList.appGroup)) {
+		if (app_list_data->appGroup &&
+		    utf8_string_to_rail_string(app_list_data->appGroup, &delete_app_list.appGroup)) {
 			deleteAppList.flags |= RDPAPPLIST_FIELD_GROUP;
 		}
-		peerCtx->applist_server_context->DeleteApplicationList(peerCtx->applist_server_context, &deleteAppList);
+		applist_ctx->DeleteApplicationList(applist_ctx, &delete_app_list);
 	Exit_deletePath:
-		if (deleteAppList.appId.string)
-			free(deleteAppList.appId.string);
-		if (deleteAppList.appGroup.string)
-			free(deleteAppList.appGroup.string);
+		free(delete_app_list.appId.string);
+		free(delete_app_list.appGroup.string);
 	} else if (app_list_data->deleteAppProvider) {
-		RDPAPPLIST_DELETE_APPLIST_PROVIDER_PDU deleteAppListProvider = {};
-		deleteAppListProvider.flags = RDPAPPLIST_FIELD_PROVIDER;
-		if (app_list_data->appProvider && // appProvider is required.
-			utf8_string_to_rail_string(app_list_data->appProvider, &deleteAppListProvider.appListProviderName))
-			peerCtx->applist_server_context->DeleteApplicationListProvider(peerCtx->applist_server_context, &deleteAppListProvider);
-		if (deleteAppListProvider.appListProviderName.string)
-			free(deleteAppListProvider.appListProviderName.string);
+		RDPAPPLIST_DELETE_APPLIST_PROVIDER_PDU del_provider = {};
+
+		del_provider.flags = RDPAPPLIST_FIELD_PROVIDER;
+		if (app_list_data->appProvider &&
+		    utf8_string_to_rail_string(app_list_data->appProvider, &del_provider.appListProviderName))
+			applist_ctx->DeleteApplicationListProvider(applistctx, &del_provider);

```

Comment:
Shouldn't be "applist_ctx" ?

Reply:
Correct. Sorry, I'm building against upstream FreeRDP and don't have applist support, so didn't get a build failure. I'll push a fix shortly.